### DISCUSSION TITLE
Why does tokio::task::Builder::spawn return a Result?

### DISCUSSION THREAD

**Discussion by hirschenberger** [tokio-rs/tokio#6946] [Q&A]
`tokio::spawn` does not and it seems the Err variant is not returned anywhere:

https://github.com/tokio-rs/tokio/blob/master/tokio/src/task/builder.rs#L93

**Comment by hawkw** [comment]
I believe this is partially for consistency with the standard library's `std::thread::spawn` (which is infallible), and `std::thread::Builder::spawn` (which similarly returns a result if a thread could not be created). I think at one point in time, this function would return an error if there were no workers in the pool and a worker thread could not be started, but that code path was eventually made infallible.
